src/sentry/models/group.py

```diff
@@ -45,6 +54,28 @@ class GroupStatus(object):
 class GroupManager(BaseManager):
     use_for_related_fields = True
 
+    def by_qualified_short_id(self, org, short_id):
+        match = _short_id_re.match(short_id.strip())
+        if match is None:
+            return
+        from sentry.models import Project
+        callsign, id = match.groups()
+        callsign = callsign.upper()
+        try:
+            project = Project.objects.get(
+                organization=org,
+                callsign=callsign
+            )
+        except Project.DoesNotExist:
+            return
+        try:
+            return Group.objects.get(
```

Comment:
You can just do this in one query instead of fetching the Project first, then Group.

```python
return Group.objects.get(
    project__organization=org,
    project__callsign=callsign,
    short_id=base36_decode(id),
)
```